pkgs/by-name/li/limo/package.nix

```diff
@@ -0,0 +1,69 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+
+  cmake,
+  pkg-config,
+
+  jsoncpp,
+  libarchive,
+  libcpr,
+  libloot,
+  pugixml,
+  unrar, # has an unfree license
+
+  libsForQt5,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "limo";
+  version = "1.0.6";
+
+  src = fetchFromGitHub {
+    owner = "limo-app";
+    repo = "limo";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-eYX6CxkSnTWbltrhp1QLwnlghy7V+1lzyvcwfWoQFB8=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    libsForQt5.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    jsoncpp
+    libarchive
+    libcpr
+    libloot
+    pugixml
+    unrar
+
+    libsForQt5.qtbase
+    libsForQt5.qtsvg
+    libsForQt5.qtwayland
+  ];
+
+  cmakeFlags = [
+    (lib.cmakeFeature "LIBUNRAR_INCLUDE_DIR" "${lib.getDev unrar}/include/unrar")
+    (lib.cmakeFeature "LIBUNRAR_PATH" "unrar")
+  ];
+
+  postInstall = ''
+    install -Dm644 ../flatpak/io.github.limo_app.limo.png -t $out/share/icons/hicolor/512x512/apps
+    install -Dm644 ../flatpak/io.github.limo_app.limo.desktop -t $out/share/applications
+  '';
```

Comment:
Pushed a change also installing the metadata file.

https://github.com/limo-app/limo/blob/13ab60d2ae676c3529807ba65417f9fa84b3dc33/src/ui/addappdialog.cpp#L90-L101
^ This is the place where `steam_app_configs` is loaded. I haven't been able to test how this works.

Reply:
`steam_app_configs` seems to be a set of default configs for well-known steam games.

My interpretation of `sfs::path config_path(is_flatpak_ ? "/app/share/steam_app_configs" : "steam_app_configs");` is that the app tries to use a relative path when it is not a flatpak bulid. I suspect there's a similar behavior for the in-app icons leading to https://github.com/NixOS/nixpkgs/pull/342877#issuecomment-2397133222

I think we'll need some patches (ideally upstream) before we can properly install them to `$out/share`?

Not sure how best to tell if it is working, but I don't think this should block initially adding the package.

---

For the icons specifically, they seem to be referenced by this stylesheet https://github.com/limo-app/limo/blob/13ab60d2ae676c3529807ba65417f9fa84b3dc33/src/ui/tagcheckbox.h#L24-L43 and also by the [icons resource file](https://github.com/limo-app/limo/blob/13ab60d2ae676c3529807ba65417f9fa84b3dc33/resources/icons.qrc#L4). The resource file is a CMake source, however I don't think the icons are...

Adding them with this patch didn't seem to help though:

<details><summary>Patch</summary>
<p>

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1c6d35e..4d6a0dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,6 +53,9 @@ endif()

 set(PROJECT_SOURCES
         resources/icons.qrc
+        resources/logo.png
+        resources/filter_accept.svg
+        resources/filter_reject.svg
         src/core/addmodinfo.h
         src/core/appinfo.h
         src/core/autotag.cpp
```

</p>
</details> 

Again, while it'd be nice to have the icons working, I don't think it should block this initial PR.

cc @limo-app in case the upstream author has any insights.

Reply:
I believe there are a lot of icons that are supposed to come from the user's icon theme.
You can see all of the path searched by running `strace Limo`

the `breeze-icons` package seems to have most of the needed icons, however I wasn't able to patch it in. I even tried just changing my actual theme using home-manager, but it didn't seem to work... I don't know too much about theming logic...

Reply:
Yep, looking at the [strace log](https://gist.githubusercontent.com/MattSturgeon/397e49829bf2b8ac95f1c51c69139e32/raw/34a83d25a42ab8f9cbf09ba3c0e9c28f296c7bad/strace.log) it does seem to be looking up icon themes for a ton of stuff.

No matches for `.qrc`, `filter_`, or `logo.png` though, so I guess that must all be statically compiled? I had initially assumed `filter_accept.svg` and `filter_reject.svg` should show up in the checkboxes highlighted by https://github.com/NixOS/nixpkgs/pull/342877#issuecomment-2397133222, however looking at the [screenshots on flathub](https://flathub.org/apps/io.github.limo_app.limo), it seems that should be normal icons from the icon theme.

> the `breeze-icons` package seems to have most of the needed icons, however I wasn't able to patch it in. I even tried just changing my actual theme using home-manager, but it didn't seem to work... I don't know too much about theming logic...

I also tried `nix profile install nixpkgs#kdePackages.breeze-icons` however this didn't help. My understanding is that an icon theme being installed should be enough for an app to use its icons, so I'm not sure what's going on here...

Maybe someone in the @NixOS/qt-kde team will have an idea? However I still think we can postpone resolving the issue for a follow up PR if needed.

Reply:
I think we should wait. Some central functionality is quite confusing without icons.

Reply:
Apologies for the late reply, i was away for a while.

> I believe there are also a lot of icons that are supposed to come from the user's icon theme.

All icons, except for *filter_accept.svg*, *filter_reject.svg* and *logo.png* from the *resources* directory, are supposed to be taken from the user's icon theme by calling [QIcon::fromTheme](https://doc.qt.io/qt-5/qicon.html#fromTheme). I am using KDE's Breeze theme which comes with every icon needed for the app. In my testing, the icons also worked for Ubuntu 24.04 with GNOME.

> is that the app tries to use a relative path when it is not a flatpak bulid.

This is correct. I expected that almost everybody would use the flatpak version and therefore did not put much thought into where additional resources should be installed to otherwise.

I will update the CMake file with non flatpak installation instructions for the next update. If it helps i could also add a list of every icon used or add a directory containing the icons themselves.
Please note that Limo is currently still using Qt 5, in case that causes any issues. I will most likely make the switch to Qt 6 in the update after the next update.

Reply:
Just to check back in, the latest revision of this PR (using 1.0.8.2) is still having issues with most icons.

I did note that _some_ icons are working (such as the file browser icon):

<details><summary>Screenshot</summary>
<p>

![image](https://github.com/user-attachments/assets/bf85f752-51cb-4714-970e-51a24897d39a)

</p>
</details> 

I wonder if this warning is at all related? :thinking: 

```
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
```

Running with `QT_QPA_PLATFORM=wayland` does get subtly different behaviour, e.g. the window is drawn with a CSD titlebar complete with limo logo. However The other missing icons still seem to be missing...

<details><summary>Screenshot with QT_QPA_PLATFORM</summary>
<p>

![image](https://github.com/user-attachments/assets/eca7a6af-13f9-44eb-9341-11511eff2ae0)

</p>
</details> 

I still think it is best to get _something_ merged and then iterate on the problem with future PRs in nixpkgs and/or upstream.